src/resources/object.rs

```diff
@@ -398,7 +398,11 @@ impl Object {
                 .send()
                 .await;
             let response = match response {
-                Ok(r) => r,
+                Ok(r) if r.status() == 200 => r,
+                Ok(r) => {
+                    let e = Error::Google(r.json().await.ok()?);
```

Comment:
Should we return `None` here when the JSON parsing fails? Wouldn't it make more sense to return an error?

Reply:
good idea. I think the entire batch of comments can be summarized as me not looking at the useful `From` implementations 😅